src/goPackages.ts

```diff
@@ -277,10 +277,10 @@ export function getNonVendorPackages(folderPath: string): Promise<string[]> {
 // see: https://golang.org/doc/go1.4#internalpackages
 // see: https://golang.org/s/go14internal
 function isAllowToImportPackage(toDirPath: string, currentWorkspace: string, pkgPath: string) {
-	let internalPkgFound = pkgPath.match(/\/internal\/|\/internal$/);
+	let internalPkgFound = pkgPath.match(/(^|\/)internal(\/|$)/);
 	if (internalPkgFound) {
 		let rootProjectForInternalPkg = path.join(currentWorkspace, pkgPath.substr(0, internalPkgFound.index));
-		return toDirPath.startsWith(rootProjectForInternalPkg + path.sep) || toDirPath === rootProjectForInternalPkg;
+		return (rootProjectForInternalPkg !== currentWorkspace && toDirPath.startsWith(rootProjectForInternalPkg + path.sep)) || toDirPath === rootProjectForInternalPkg;
```

Comment:
Wouldn't this return `true` for pkgPath which is say `internal/something` and user is working on a file directly under `GOPATH/src`?

Reply:
Impossible to work under `GOPATH/src`

Ahh.. yes.. correct.